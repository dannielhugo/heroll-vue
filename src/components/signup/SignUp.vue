<template>
  <div class="sign-in">
    <div class="sign-in__bg">
      <el-image class="sign-in__bg__img" fit="cover" :src="url" />
      <div class="sign-in__bg__credits">
        Photo by
        <a
          href="https://unsplash.com/@nickhamze?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
          >Nick Hamze</a
        >
        on
        <a
          href="https://unsplash.com/s/photos/video-game?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText"
          >Unsplash</a
        >
      </div>
    </div>
    <div class="sign-in__card">
      <el-card class="sign-in__card__el">
        <h1>Heroll - Sign Up</h1>
        <el-button
          type="primary"
          data-testid="signup"
          size="large"
          :icon="UserFilled"
          :loading="loading"
          @click="$emit('signup')"
        >
          Sign Up With Google
        </el-button>
      </el-card>
    </div>
  </div>
</template>

<script setup lang="ts">
import { UserFilled } from '@element-plus/icons-vue';

import url from '@/assets/signin-bg.jpg';
defineEmits(['signup']);
defineProps(['loading']);
</script>

<style scoped lang="scss">
.sign-in {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  height: 80vh;

  &__bg {
    position: relative;
    height: 400px;
    display: inline-block;

    &__img {
      width: 600px;
      height: 400px;
    }

    &__credits {
      position: absolute;
      left: 5px;
      bottom: 5px;
      color: var(--el-color-info-light-9);
      a {
        color: var(--el-color-info-light-9);
      }
    }
  }

  &__card {
    display: inline-block;
    height: 400px;
    vertical-align: top;

    &__el {
      width: 400px;
      height: 400px;
      text-align: center;

      h1 {
        margin-bottom: 50px;
      }
    }
  }
}
</style>
