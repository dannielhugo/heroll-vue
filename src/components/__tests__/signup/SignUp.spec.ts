import { mount } from '@vue/test-utils';
import ElementPlus from 'element-plus';
import SignUp from '../../signup/SignUp.vue';

describe('SignUp', () => {
  test('should mount component', async () => {
    expect(SignUp).toBeTruthy();

    const wrapper = mount(SignUp, {
      props: {
        loading: false,
      },
      global: {
        plugins: [ElementPlus],
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should emit signup event', () => {
    const wrapper = mount(SignUp, {
      props: {
        loading: false,
      },
      global: {
        plugins: [ElementPlus],
      }
    });

    wrapper.find('[data-testid=signup]').trigger('click');

    expect(wrapper.emitted()).toHaveProperty('signup')
  });
});