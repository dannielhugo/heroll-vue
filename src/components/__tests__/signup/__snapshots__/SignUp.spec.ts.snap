// Vitest Snapshot v1

exports[`SignUp > should mount component 1`] = `
"<div class=\\"sign-in\\" data-v-facc1a62=\\"\\">
  <div class=\\"sign-in__bg\\" data-v-facc1a62=\\"\\">
    <div class=\\"el-image sign-in__bg__img\\" data-v-facc1a62=\\"\\">
      <!--v-if-->
      <div class=\\"el-image__wrapper\\">
        <div class=\\"el-image__placeholder\\"></div>
      </div>
      <!--v-if-->
    </div>
    <div class=\\"sign-in__bg__credits\\" data-v-facc1a62=\\"\\"> Photo by <a href=\\"https://unsplash.com/@nickhamze?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText\\" data-v-facc1a62=\\"\\">Nick Hamze</a> on <a href=\\"https://unsplash.com/s/photos/video-game?utm_source=unsplash&amp;utm_medium=referral&amp;utm_content=creditCopyText\\" data-v-facc1a62=\\"\\">Unsplash</a></div>
  </div>
  <div class=\\"sign-in__card\\" data-v-facc1a62=\\"\\">
    <div class=\\"el-card is-always-shadow sign-in__card__el\\" data-v-facc1a62=\\"\\">
      <!--v-if-->
      <div class=\\"el-card__body\\" style=\\"\\">
        <h1 data-v-facc1a62=\\"\\">Heroll - Sign Up</h1><button class=\\"el-button el-button--primary el-button--large\\" aria-disabled=\\"false\\" type=\\"button\\" data-testid=\\"signup\\" data-v-facc1a62=\\"\\"><i class=\\"el-icon\\"><svg viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\">
              <path fill=\\"currentColor\\" d=\\"M288 320a224 224 0 1 0 448 0 224 224 0 1 0-448 0zm544 608H160a32 32 0 0 1-32-32v-96a160 160 0 0 1 160-160h448a160 160 0 0 1 160 160v96a32 32 0 0 1-32 32z\\"></path>
            </svg></i><span class=\\"\\"> Sign Up With Google </span></button>
      </div>
    </div>
  </div>
</div>"
`;
